### PR TITLE
feat: support optional expiresIn param in token creation [AHOY-2470]

### PR DESCRIPTION
**Pull request by budimirbudimir**
## Summary

- Support optional `expiresIn` parameter in `CreatePersonalAccessTokenProps`.
- Support optional `expiresAt` parameter in `PersonalAccessTokenProp.sys`.

## Description

- Update 2 out-of-sync types to support new CMA access tokens TTL features.
- Test `personalAccessToken.create` and `personalAccessToken.get` methods on the plain client
- Test `createPersonalAccessToken` and `getPersonalAccessToken` methods on the default client

## Motivation and Context

We are updating the client types to reflect the API changes.
In the follow up pull requests, additional changes will be introduced, moving away from Personal Access Tokens (PATs) and using new entity of unified Access Tokens (PAT, CLI, OAuth) will be introduced.
For now, though, we are just updating PAT types.

## Checklist (check all before merging)

- [ ] Both unit and integration tests are passing
- [ ] There are no breaking changes
- [ ] Changes are reflected in the documentation

When adding a new method:

- [ ] The new method is exported through the default and plain CMA client
- [ ] All new public types are exported from `./lib/export-types.ts`
- [ ] Added a unit test for the new method
- [ ] Added an integration test for the new method
- [ ] The new method is added to the documentation

### CODE REVIEW

**Review by piankovev — APPROVED** [review]
Looks good to me 👍

File: lib/entities/personal-access-token.ts [hunk 1]
```diff
@@ -5,14 +5,16 @@ import { wrapCollection } from '../common-utils'
 import { MetaSysProps, DefaultElements, MakeRequest } from '../common-types'
 
 export type PersonalAccessTokenProp = {
-  sys: MetaSysProps
+  sys: MetaSysProps & { expiresAt?: number }
```
Comment (Annmary12) [inline lib/entities/personal-access-token.ts:8]:
```suggestion
  sys: MetaSysProps & { expiresAt?: string }
```

Comment (budimirbudimir) [reply lib/entities/personal-access-token.ts:8]: @Annmary12 I actually believe it should be a number here, right?
As we're sending the number to the API as well:

Screenshot 2023-09-01 at 15 22 51

Comment (budimirbudimir) [reply lib/entities/personal-access-token.ts:8]: Also, in `GenerateCMATokenDialog` on lines 148-150, we are also setting the following payload as number:

Screenshot 2023-09-01 at 15 25 53

Comment (Annmary12) [reply lib/entities/personal-access-token.ts:8]: I was not referring to `expiresIn`, it is `expiresAt` that I highlighted which is a date string

Comment (budimirbudimir) [reply lib/entities/personal-access-token.ts:8]: Sorry, you are correct, not sure why I went onto `expiresIn`. 🤦 
I'll see to fix this. 👍

**Review by Annmary12 — APPROVED** [review]
🎉

**Pull request merged**